### DISCUSSION TITLE
Accessing non-existent property 'lineno' of module exports inside circular dependency如何解决呢

### DISCUSSION THREAD

**Discussion by penndu** [volantis-x/hexo-theme-volantis#738] [Q&A] [answered]
报错代码如下：
```
(node:7973) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:7973) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:7973) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:7973) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:7973) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:7973) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
```

**Comment by MHuiG** [comment]
是warning不是报错



 ----------回复的邮件信息----------
 ***@***.***&gt; 于2022年4月14日(周四) 晚上10:41写道:

  

 
报错代码如下：
 (node:7973) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency (Use `node --trace-warnings ...` to show where the warning was created) (node:7973) Warning: Accessing non-existent property 'column' of module exports inside circular dependency (node:7973) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency (node:7973) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency (node:7973) Warning: Accessing non-existent property 'column' of module exports inside circular dependency (node:7973) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency

**Reply by penndu** [reply]
> 

如果想去掉请问该如何操作，找了一些方法：降低nodejs的版本，修改node_modules/stylus/lib/nodes/index.js不生效。

**Comment by MHuiG** [comment]
我也想知道。



 ----------回复的邮件信息----------
 ***@***.***&gt; 于2022年4月14日(周四) 晚上10:50写道:

  

   
如果想去掉请问该如何操作，找了一些方法：降低版本，修改index.js不生效。

**Reply by penndu** [reply]
> 我也想知道。
> […](#)
> ----------回复的邮件信息---------- ***@***.***&gt; 于2022年4月14日(周四) 晚上10:50写道: 如果想去掉请问该如何操作，找了一些方法：降低版本，修改index.js不生效。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

![](https://npm.elemecdn.com/sticker-heo@0.17.2/Sticker-100/%E6%8D%82%E8%84%B8.png)

**Comment by MHuiG** [comment]
https://github.com/volantis-x/demo/blob/8022d0f4f1c954b0debc609de5b301f72bcaf5a5/package.json#L14-L18

https://github.com/stylus/nib/issues/354

node 16 npm 8 修改 package.json ，demo 现在是没问题的

```json
  "overrides" : {
    "nib" : {
      "stylus" : "0.54.8"
    }
  },
```

**Comment by MHuiG** [comment]
更简单的方法：

```bash
npm un hexo-renderer-stylus
npm i @copoko/hexo-renderer-stylus -S
```

**Comment by MHuiG** [comment, marked as answer]
https://github.com/stylus/nib/pull/359

https://www.npmjs.com/package/nib


![image](https://user-images.githubusercontent.com/38281581/168723552-47b2247b-72fe-48a1-b303-46411d0f4a03.png)


就在几分钟前 stylus/nib 发布了新的版本解决了这个问题

https://github.com/hexojs/hexo-renderer-stylus/pull/96

删掉 package-lock.json 重装一下 hexo-renderer-stylus 即可解决
